### PR TITLE
add react native support, add in read/write/delete/exists methods

### PR DESCRIPTION
**Pull request by weixingsun**
will add test cases soon

### CODE REVIEW

File: src/15utility.js [hunk 1]
```diff
@@ -272,6 +272,10 @@ utils.isCordova = (function(){
   return (typeof cordova === 'object')
 })();
 
+utils.isReactNative = (function(){
+  return (typeof require('react-native') === 'object')
```
Comment (mathiasrw) [inline src/15utility.js]: Hmm... what is "require" is not defined?

File: src/15utility.js [hunk 2]
```diff
@@ -339,7 +343,14 @@ var loadFile = utils.loadFile = function(path, asy, success, error) {
                 }
             }
         }
-        //*/
```
Comment (mathiasrw) [inline src/15utility.js]: The end of the comment is removed (used to strip out things never used for the browser) - please add again (after the code you added)

File: src/15utility.js [hunk 3]
```diff
@@ -492,9 +503,17 @@ var loadBinaryFile = utils.loadBinaryFile = function(path, asy, success, error)
                 success(arr.join(""));
             }
         }
-		//*/
```
Comment (mathiasrw) [inline src/15utility.js]: Please re-add

File: src/15utility.js [hunk 4]
```diff
@@ -541,7 +560,14 @@ var removeFile = utils.removeFile = function(path,cb) {
                 cb && cb(); // jshint ignore:line
             });
         });
-      	//*/
```
Comment (mathiasrw) [inline src/15utility.js]: Please re-add

File: src/15utility.js [hunk 5]
```diff
@@ -571,7 +605,15 @@ var fileExists = utils.fileExists = function(path,cb){
                 cb(false);
             });
         });
-        //*/
```
Comment (mathiasrw) [inline src/15utility.js]: Please re-add

**Comment by mathiasrw** [comment]
Would you like some help to get the PR finalised?

**Comment by mathiasrw** [comment]
Hey @weixingsun  

is there anything I can do to help out?

**Comment by weixingsun** [comment]
@mathiasrw sorry for the delay, I will do npm test on my code, and sorry for //*/,  will re-add them.

File: src/15utility.js (hunk 1 above)
Comment (weixingsun) [inline src/15utility.js]: simply check if package 'react-native' is available, please let me know if you have another way like 'isCordova', but I cannot find definition of object 'cordova'.

**Comment by weixingsun** [comment]
@mathiasrw please help.
I got some errors when did npm test:

alasql weixingsun$ npm test

> alasql@0.3.4 test /Users/weixingsun/GitHub/alasql
> gulp && cd test && mocha . --reporter dot

[14:04:20] Using gulpfile ~/GitHub/alasql/gulpfile.js
[14:04:20] Starting 'js-merge'...
[14:04:20] Starting 'js-merge-worker'...
[14:04:20] Starting 'plugin-prolog'...
[14:04:20] Finished 'plugin-prolog' after 1.86 ms
[14:04:20] Starting 'plugin-plugins'...
[14:04:20] Finished 'plugin-plugins' after 2.33 ms
[14:04:20] Starting 'typescript'...
[14:04:20] Finished 'typescript' after 1 ms
[14:04:20] Finished 'js-merge-worker' after 390 ms

events.js:160
      throw er; // Unhandled 'error' event
      ^
GulpUglifyError: unable to minify JavaScript
    at createError (/Users/weixingsun/GitHub/alasql/node_modules/gulp-uglify/lib/create-error.js:6:14)
    at wrapper (/Users/weixingsun/GitHub/alasql/node_modules/gulp-uglify/node_modules/lodash/_createHybrid.js:87:15)
    at trycatch (/Users/weixingsun/GitHub/alasql/node_modules/gulp-uglify/minifier.js:26:12)
    at DestroyableTransform.minify [as _transform] (/Users/weixingsun/GitHub/alasql/node_modules/gulp-uglify/minifier.js:76:19)
    at DestroyableTransform.Transform._read (/Users/weixingsun/GitHub/alasql/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:159:10)
    at DestroyableTransform.Transform._write (/Users/weixingsun/GitHub/alasql/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:147:83)
    at doWrite (/Users/weixingsun/GitHub/alasql/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:345:64)
    at writeOrBuffer (/Users/weixingsun/GitHub/alasql/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:334:5)
    at DestroyableTransform.Writable.write (/Users/weixingsun/GitHub/alasql/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:271:11)
    at Transform.ondata (_stream_readable.js:555:20)
npm ERR! Test failed.  See above for more details.

File: src/15utility.js [hunk 6]
```diff
@@ -339,7 +343,15 @@ var loadFile = utils.loadFile = function(path, asy, success, error) {
                 }
             }
         }
-        //*/
+	} else if(utils.isReactNative) {
+        /* If ReactNative */
```
Comment (mathiasrw) [inline src/15utility.js]: The problem with uglyfy2 is that having start comments within a section that is stripped from the browserversion is not supported. Just change to `// if reactiveNative` to make it work. 

Second input: As this is also stated in the `If` just above, I recommend ` /* If ReactNative */`gets removed

File: src/15utility.js [hunk 7]
```diff
@@ -492,9 +504,18 @@ var loadBinaryFile = utils.loadBinaryFile = function(path, asy, success, error)
                 success(arr.join(""));
             }
         }
-		//*/
+	} else if(utils.isReactNative) {
+        /* If ReactNative */
```
Comment (mathiasrw) [inline src/15utility.js]: As this is also stated in the `If` just above, I recommend ` /* If ReactNative */`gets removed

File: src/15utility.js [hunk 8]
```diff
@@ -541,7 +562,15 @@ var removeFile = utils.removeFile = function(path,cb) {
                 cb && cb(); // jshint ignore:line
             });
         });
-      	//*/
+	} else if(utils.isReactNative) {
+        /* If ReactNative */
```
Comment (mathiasrw) [inline src/15utility.js]: As this is also stated in the `If` just above, I recommend ` /* If ReactNative */`gets removed

File: src/15utility.js [hunk 9]
```diff
@@ -553,6 +582,14 @@ var deleteFile = utils.deleteFile = function(path,cb){
     if(utils.isNode) {
         var fs = require('fs');
         fs.unlink(path, cb);
+	} else if(utils.isReactNative) {
+        /* If ReactNative */
```
Comment (mathiasrw) [inline src/15utility.js]: As this is also stated in the `If` just above, I recommend ` /* If ReactNative */`gets removed

File: src/15utility.js [hunk 10]
```diff
@@ -571,7 +608,15 @@ var fileExists = utils.fileExists = function(path,cb){
                 cb(false);
             });
         });
-        //*/
+	} else if(utils.isReactNative) {
+        /* If ReactNative */
```
Comment (mathiasrw) [inline src/15utility.js]: As this is also stated in the `If` just above, I recommend ` /* If ReactNative */`gets removed

**Comment by mathiasrw** [comment]
Woohoooo !

**Pull request merged**